src/Client/Pages/Identity/Roles/RolePermissions.razor.cs

```diff
@@ -0,0 +1,102 @@
+﻿using FSH.BlazorWebAssembly.Client.Infrastructure.ApiClient;
+using FSH.BlazorWebAssembly.Client.Shared;
+using FSH.BlazorWebAssembly.Shared.Authorization;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+using Mapster;
+
+namespace FSH.BlazorWebAssembly.Client.Pages.Identity.Roles;
+
+public partial class RolePermissions
+{
+    [CascadingParameter]
+    protected Task<AuthenticationState> AuthState { get; set; } = default!;
+    [Inject]
+    protected IAuthorizationService AuthService { get; set; } = default!;
+    [Parameter]
+    public string? Id { get; set; }
+    [Parameter]
+    public string? Title { get; set; }
+    [Parameter]
+    public string? Description { get; set; }
+    [Inject]
+    protected IRolesClient RolesClient { get; set; } = default!;
+    public List<PermissionDto> RolePermissionsList { get; set; } = new();
+
+    private PermissionDto _permission = new();
+    private string _searchString = string.Empty;
+    private bool _dense = false;
+    private bool _striped = true;
+    private bool _bordered = false;
+
+    private bool _canEditRoleClaims;
+    private bool _canSearchRoleClaims;
+    private bool _loaded;
+
+    protected override async Task OnInitializedAsync()
+    {
+        var state = await AuthState;
+        _canEditRoleClaims = (await AuthService.AuthorizeAsync(state.User, FSHPermissions.RoleClaims.Edit)).Succeeded;
+        _canSearchRoleClaims = (await AuthService.AuthorizeAsync(state.User, FSHPermissions.RoleClaims.View)).Succeeded;
+
+        var rolePermissions = new List<PermissionDto>();
+
+        if (await ApiHelper.ExecuteCallGuardedAsync(
+                () => RolesClient.GetByIdAsync(Id), Snackbar)
+            is RoleDto role)
+        {
+            Title = role.Name;
+            Description = string.Format(_localizer["Manage {0}'s Permissions"], role.Name);
+
+            if (await ApiHelper.ExecuteCallGuardedAsync(
+                    () => RolesClient.GetPermissionsAsync(role.Id), Snackbar)
+                is ICollection<PermissionDto> response)
+            {
+                rolePermissions = response.ToList();
+            }
+        }
+
+        var allPermissions = DefaultPermissions.Admin;
+        allPermissions.AddRange(DefaultPermissions.Root);
+        var result = allPermissions.Select(x => new PermissionDto()
+        {
+            Permission = x
+        }).ToList();
+
+        foreach (var permission in result)
+        {
+            if (rolePermissions.Select(z => z.Permission).Contains(permission.Permission))
+                permission.Enabled = true;
```

Comment:
Where is this enabled coming from? PermissionDto doesn't have an 'Enabled' property?

Reply:
PermissionDto has enbaled property yes, what doesnt have is the UpdatePermissionRequest. I Think we should keep it so it is equal to another Client apps

Reply:
I don't see it (https://github.com/fullstackhero/dotnet-webapi-boilerplate/blob/main/src/Core/Application/Identity/Roles/PermissionDto.cs).

It shouldn't be there anyway...

Reply:
it´s only on my PR, so create a new Dto only on the client side?

Reply:
Yeah, it's done on other places, like for instance AuditLogs.Razor.cs there you have RelatedAuditTrail which is derived from AuditDto.